src/OrchardCore.Modules/OrchardCore.Layers/Drivers/LayerMetadataWelder.cs

```diff
@@ -65,26 +60,38 @@ public override async Task<IDisplayResult> UpdateAsync(ContentItem model, Update
 
         await context.Updater.TryUpdateModelAsync(viewModel, Prefix);
 
-        if (viewModel.LayerMetadata == null)
+        if (string.IsNullOrEmpty(viewModel.Title))
         {
-            return null;
+            context.Updater.ModelState.AddModelError(Prefix, nameof(viewModel.Title), S["The Title field is required"]);
         }
 
-        if (string.IsNullOrEmpty(viewModel.LayerMetadata.Zone))
+        if (string.IsNullOrEmpty(viewModel.Zone))
         {
-            context.Updater.ModelState.AddModelError(Prefix, "LayerMetadata.Zone", S["Zone is missing"]);
+            context.Updater.ModelState.AddModelError(Prefix, nameof(viewModel.Zone), S["The Zone field is required"]);
         }
 
-        if (string.IsNullOrEmpty(viewModel.LayerMetadata.Layer))
+        if (string.IsNullOrEmpty(viewModel.Layer))
         {
-            context.Updater.ModelState.AddModelError(Prefix, "LayerMetadata.Layer", S["Layer is missing"]);
+            context.Updater.ModelState.AddModelError(Prefix, nameof(viewModel.Layer), S["The Layer field is required"]);
         }
```

Comment:
These can be shorter attributes on the view model instead.

Reply:
I rather see the validation in the driver like we do most of the time. It's easier to see the rule when looking at the driver which is the only thing that use that view model.

Reply:
And it's really easy to see all the `Required` attributes on the view model as well :). You're reinventing the wheel here. For complex validation, sure, we need this. But not for required checks. The attributes also let the tag helpers generate `required` attributes in HTML.

Reply:
In the driver, I can ensure that the name is valid which isn't doable in a vm. I wouldn't want some validation in the VM and other in the DD. + in most of places we use DD validation over DataAnnotation.

Reply:
I don't agree but I also don't see any point debating this further.